vendor/github.com/weaveworks/go-checkpoint/checkpoint.go

```diff
@@ -284,6 +291,31 @@ func checkResult(r io.Reader) (*CheckResponse, error) {
 	return &result, nil
 }
 
+// getSystemUUID returns the base64 encoded, scrypt hashed contents of
+// /sys/class/dmi/id/product_uuid (with some salt)
+func getSystemUUID() (string, error) {
+	uuid, err := ioutil.ReadFile("/sys/class/dmi/id/product_uuid")
+	if err != nil {
+		return "", err
+	}
+	if len(uuid) <= 0 {
+		return "", fmt.Errorf("Empty system uuid")
+	}
```

Comment:
Why is an empty uuid file an error? Shouldn't that be the same as not finding one?